pkgs/development/interpreters/cg3/default.nix

```diff
@@ -0,0 +1,61 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, runCommand
+, dieHook
+, cmake
+, icu
+, boost
+}:
+
+let cg3 = stdenv.mkDerivation rec {
+  pname = "cg3";
+  version = "1.3.7";
+
+  src = fetchFromGitHub {
+    owner = "GrammarSoft";
+    repo = "${pname}";
+    rev = "v${version}";
+    sha256 = "Ena3dGoZsXOIY6mbvnfI0H7QqRifoxWIBKQrK3yQSmY=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  buildInputs = [
+    icu
+    boost
+  ];
+
+  doCheck = true;
+
+  passthru.tests.minimal = runCommand "${pname}-test" {
+      buildInputs = [
+        cg3
+        dieHook
+      ];
+    } ''
+      echo 'DELIMITERS = "."; ADD (tag) (*);' >grammar.cg3
+      printf '"<a>"\n\t"a" tag\n\n' >want.txt
+      printf '"<a>"\n\t"a"\n\n' | vislcg3 -g grammar.cg3 >got.txt
+      diff -s want.txt got.txt || die "Grammar application did not produce expected parse"
+      touch $out
+    '';
+
+
+  # TODO, consider optionals:
+  # - Enable tcmalloc unless darwin?
+  # - Enable python bindings?
+
+  meta = with lib; {
+    homepage = "https://github.com/GrammarSoft/cg3";
+    description = "Constraint Grammar interpreter, compiler and applicator vislcg3";
+    maintainers = [ lib.maintainers.unhammer ];
+    license = licenses.gpl3Plus;
+    platforms = platforms.all;
+  };
+};
+
+in
+  cg3
```

Comment:
This is the same as not using `let ... in`, so you can drop both sides.

Reply:
but I refer to cg3 in buildInputs to tests (got an error without that) – is there a better way to get cg3 into test dependencies?

Reply:
Apologies, I did not notice. That is indeed the way it's used in other packages.